content/articles/continuous-integration-and-deployment-of-pipelines-with-flask-docker-and-github-actions/index.md

```diff
@@ -0,0 +1,234 @@
+
+Above are the practices that applications undergo in the industry. A skilled person in terms of all these is said to be a DevOps Engineer.
+
+### Cloning a Flask Application
+You are required to clone a Flask application from this [repository](https://github.com/Horlawhumy-dev/flask-drinksapi). This contains a simple Flask API that we will work on with this tutorial.
+
+The following are the set of commands that will walk you through the cloning.
+
+```git
+$ git clone git@github.com:Horlawhumy-dev/flask-drinksapi.git
+$ cd flask-drinksapi
+$ source venv/Scripts/activate
+$ pip install -r requirements.txt
+$ python3 app.python
+```
+
+Following these commands, you used the first one to clone from the remote repository into your local computer. Then you change the directory into the folder cloned with the following command.
+
+Every python application needs a `virtual environment` in order to activate the dependency. Inside the cloned folder, there is a folder `venv` that contains the environment and you used the third command to activate it.
+
+Furthermore, you installed the dependencies inside the `requirements.txt` file. You then started the server with the last command on the snippet above.
+
+### CI and CD Pipelines Architecture
+The continuous integration and deployment of pipelines are software development practices used in the industry to develop and maintain every application. 
+
+Developing applications from the ground up requires so many practices and keeping track of every change in order not to break the already deployed application.
+
+The continuous integration server is going to trigger the automated build test job. This is to check whether the pushed code is reliable or not. If the code is reliable, then it will be integrated, built, and sent to the deployment server. Successful notifications will be sent too and this is referred to as automation.
+
+If otherwise, the pushed codes fail to build, another notification will be sent. So that the developer can fix the bugs and then re-commit the code. The continuous integration tool will then start building the code again. This is the reason why the continuous integration and deployment stages are still considered a waterfall model.
+
+### Continuous integration and deployment with Github Actions
+Github Actions automate the build process whenever there is a push to the code repository. This helps to reduce the problem of a manual building of the jobs and always notify if the build is not complete. This
+
+This follows a workflow using a different language package file. Earlier, we cloned a Python application built with the Flask framework. Therefore, the workflow that we are going to define will be for the Python package.
+
+Navigate to [project](https://github.com/Horlawhumy-dev/flask-drinksapi), click on the `action` link, and then you should receive the image below.
+
```

Comment:
Workflows appear differently on forked repos. Direct the reader to setup the workflow on their own GitHub repository.